src/dotnet/commands/dotnet-new3/New3CommandShim.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Config;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.DotNet.Tools.New3
+{
+    internal class New3CommandShim
+    {
+        private const string HostIdentifier = "dotnetcli";
+        private const string HostVersion = "1.0.0";
+        private const string CommandName = "new3";
+
+        public static int Run(string[] args)
+        {
+            return New3Command.Run(CommandName, CreateHost(), FirstRun, args);
+        }
+
+        private static ITemplateEngineHost CreateHost()
+        {
+            var builtIns = new Dictionary<Guid, Func<Type>>
+            {
+                { new Guid("0C434DF7-E2CB-4DEE-B216-D7C58C8EB4B3"), () => typeof(RunnableProjectGenerator) },
```

Comment:
Can we make the default set defined here the defaults in DefaultTemplateEngineHost and just pass null from CLI? I think that the concern is that these GUIDs don't really mean anything to CLI and there is not a present desire for changing the default behavior of the host...

Alternatively, we would want to add a meaningful test for these configurations to ensure we're not accidentally breaking some desirable behavior. I'd personally prefer having the defaults built in, but we can do this instead...